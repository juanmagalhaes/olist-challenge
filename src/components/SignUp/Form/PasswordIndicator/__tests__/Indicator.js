import React from 'react';
import {
  Indicator,
  Indicators,
} from '../Indicator';

describe('Indicator', () => {
  const shallowWrapper = props => shallow(<Indicator {...props} />);

  test('renders', () => {
    const wrapper = shallowWrapper({ style: 'Default' });
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Indicators', () => {
  const shallowWrapper = props => shallow(<Indicators {...props} />);

  test('renders', () => {
    expect(shallowWrapper({ validCount: 0 })).toMatchSnapshot();
    expect(shallowWrapper({ validCount: 1 })).toMatchSnapshot();
    expect(shallowWrapper({ validCount: 2 })).toMatchSnapshot();
    expect(shallowWrapper({ validCount: 3 })).toMatchSnapshot();
  });
});

