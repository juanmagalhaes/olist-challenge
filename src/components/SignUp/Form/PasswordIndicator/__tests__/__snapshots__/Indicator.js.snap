// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indicator renders 1`] = `
<div
  className="indicator indicatorDefault"
/>
`;

exports[`Indicators renders 1`] = `
<div
  className="indicators"
>
  <Indicator
    key="0"
    style="Default"
  />
  <Indicator
    key="1"
    style="Default"
  />
  <Indicator
    key="2"
    style="Default"
  />
</div>
`;

exports[`Indicators renders 2`] = `
<div
  className="indicators"
>
  <Indicator
    key="0"
    style="Danger"
  />
  <Indicator
    key="1"
    style="Default"
  />
  <Indicator
    key="2"
    style="Default"
  />
</div>
`;

exports[`Indicators renders 3`] = `
<div
  className="indicators"
>
  <Indicator
    key="0"
    style="Waring"
  />
  <Indicator
    key="1"
    style="Waring"
  />
  <Indicator
    key="2"
    style="Default"
  />
</div>
`;

exports[`Indicators renders 4`] = `
<div
  className="indicators"
>
  <Indicator
    key="0"
    style="Success"
  />
  <Indicator
    key="1"
    style="Success"
  />
  <Indicator
    key="2"
    style="Success"
  />
</div>
`;
